tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+  install is done (which in this case is `apache2` itself, and all its
+  dependencies).
+
+```text
+Reading package lists... Done
+Building dependency tree... Done
+Reading state information... Done
+The following packages were automatically installed and are no longer required:
+  linux-headers-6.8.0-39 linux-headers-6.8.0-39-generic linux-image-6.8.0-39-generic linux-modules-6.8.0-39-generic linux-tools-6.8.0-39
+  linux-tools-6.8.0-39-generic
+Use 'sudo apt autoremove' to remove them.
+The following additional packages will be installed:
+  apache2-bin apache2-data apache2-utils libapr1t64 libaprutil1-dbd-sqlite3 libaprutil1-ldap libaprutil1t64 liblua5.4-0 ssl-cert
+Suggested packages:
+  apache2-doc apache2-suexec-pristine | apache2-suexec-custom www-browser
+The following NEW packages will be installed:
+  apache2 apache2-bin apache2-data apache2-utils libapr1t64 libaprutil1-dbd-sqlite3 libaprutil1-ldap libaprutil1t64 liblua5.4-0 ssl-cert
+0 upgraded, 10 newly installed, 0 to remove and 0 not upgraded.
+Need to get 2083 kB of archives.
+After this operation, 8094 kB of additional disk space will be used.
+Do you want to continue? [Y/n] 
+```
+
+Let's try and make sense of this output. 
+
+#### Types of dependencies
+
+The relationship between a package and any other packages follows the
+[Debian policy on binary dependencies](https://www.debian.org/doc/debian-policy/ch-relationships.html#binary-dependencies-depends-recommends-suggests-enhances-pre-depends),
+which we'll briefly look at here. The most common ones you might come
+across are: `depends`, `recommends`, and `suggests`(although there are
+others!), so we'll take a look at these three.
+
+- **depends**: Absolutely required, the package won't work without it.
+- **recommends**: Strongly dependent, but not absolutely necessary (which means
+  the package will work better with it, but can still function without it)
+- **suggests**: Not needed, but may enhance the usefulness of the package in
+  some way.
+
+We can see, using `apt show`, exactly which packages fall into each of these
+categories. Let's use Apache2 as our example again:
+
+```bash
+apt show apache2
+```
+
+If we look only at the sections on dependencies, we can see that `ssl-cert` is
+a recommended package:
+
+```text
+Package: apache2
+Version: 2.4.58-1ubuntu8.4
+[...]
+Provides: httpd, httpd-cgi
+Pre-Depends: init-system-helpers (>= 1.54~)
+Depends: apache2-bin (= 2.4.58-1ubuntu8.4), apache2-data (= 2.4.58-1ubuntu8.4), apache2-utils (= 2.4.58-1ubuntu8.4), media-types, perl:any, procps
+Recommends: ssl-cert
+Suggests: apache2-doc, apache2-suexec-pristine | apache2-suexec-custom, www-browser, ufw
+[...]
+```
+
+In Ubuntu, the default configuration of `apt install` is set to install
+recommended packages alongside `depends`, so when we ran the
+`apt install apache2` command, `ssl-cert` was included in the packages
+to be installed (even though it's only recommended, and not strictly needed).
```

Comment:
Maybe include a step to show that ssl-cert is already installed? (`apt list ssl-cert`?)

Reply:
If they're following the commands/instructions exactly, a user shouldn't have ssl-cert installed yet. This whole section is just about the different dependency types and how to find out what's recommended/required/suggested etc (and a bit of an introduction in how to use flags).

Reply:
I need to make it a bit stronger in the intro to the section I think.